frontend/public/components/_dropdown.scss

```diff
@@ -263,3 +263,64 @@ $dropdown-background--hover: $color-pf-black-200; // pf-c-dropdown__menu-item--h
     display: none !important;
   }
 }
+
+/*********** The following is for the namespace-dropdown and namespace-menu components ************** */
+
+.co-namespace-project-selector {
+  .pf-c-divider {
```

Comment:
Agreed.  And if we do need overrides,

1.  We should avoid using descendant selectors and the PatternFly class names.  Can you simply pass a `className` on the component that needs the override and write a rule for it? (In this case, `<DropdownSeparator className="foo">` works).  If you can't pass a `className` on the component that needs it, a descendant selector is ok but not ideal.
2. Add a comment explaining why the override is needed.  If a descendant selector is needed, comment on why.
3. If the override is because of an upstream bug, open an issue and include it in the above comment.

Reply:
There are a couple of reasons for overwriting the PatternFly css.  All of them are cosmetic - meaning the application would be functional and understandable without the custom styles.  I added custom classnames where it was possible and added full description of why it is needed.  Changes in next commit.

Reply:
@kdoberst I think the override for the menu separator might actually be a bug in PF. Without this override, the separator isn't showing up at all, right?

Reply:
@TheRealJon You are right, the divider doesn't show at all without the override.  I also thought that it was a PF bug, but when I did some digging, the Bootstrap SCSS (`./public/node_modules/bootstrap-sass/assets/stylesheets/bootstrap/_normalize.scss`) is setting the height to 0 and this override is setting it back to the PF setting.  

To make this clear, I added this fact as comments into  _dropdown.scss - that will be in the next commit.

Reply:
This conversation wasn't marked as "Outdated", but just wanted to put a note that in the SCSS file, I'm using custom classnames when I can and have added comments for areas where I had to tap into the PF classes.